templates/pkg/resource/manager_factory.go.tpl

```diff
@@ -38,8 +39,9 @@ func (f *resourceManagerFactory) ManagerFor(
 	id ackv1alpha1.AWSAccountID,
 	region ackv1alpha1.AWSRegion,
 ) (acktypes.AWSResourceManager, error) {
+	rmId := fmt.Sprintf("%s/%s", id, region)
 	f.RLock()
-	rm, found := f.rmCache[id]
+	rm, found := f.rmCache[rmId]
 	f.RUnlock()
 
 	if found {
```

Comment:
One other "working" solution to this bug could have been to simply check for the region after finding a resourceManager.. But this solution means that we will recreate and override the resourceManager everytime we create a resource in a different region
```golang
if found && region == rm.awsRegion {
    return rm, nil
}
```